pvlib/tests/iotools/test_psm3.py

```diff
@@ -49,48 +61,72 @@ def assert_psm3_equal(header, data, expected):
     assert (data.index.tzinfo.zone == 'Etc/GMT%+d' % -header['Time Zone'])
 
 
+def delay_get_psm3(*args, **kwargs):
+    """Repeat get_psm3() call if OVER_RATE_LIMIT error occurs otherwise
+    allow HTTPError to raise. Sleep times increase by one second for
+    every OVER_RATE_LIMIT HTTPError in a row. Return (header, data) from
+    get_psm3() as normal.
+    """
+    sleep_time = 0
+    while True:
```

Comment:
I'm always leery of while loops... what about a for loop with a limit e.g. `for _ in range(max_retries)`

Reply:
I get that. I also considered using `for sleep_time in range(max_retries):` to consolidate the iteration of sleep_time and the looping. I'd say `max_retries = 5` would suffice, would you agree?